src/main/java/uk/gov/hmcts/reform/roleassignment/data/roleassignment/RoleAssignmentRepository.java

```diff
@@ -1,19 +1,29 @@
 
 package uk.gov.hmcts.reform.roleassignment.data.roleassignment;
 
-import org.springframework.data.jpa.repository.JpaRepository;
-import org.springframework.stereotype.Repository;
-
 import java.util.Set;
 import java.util.UUID;
 
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.stereotype.Repository;
+
 @Repository
 public interface RoleAssignmentRepository extends JpaRepository<RoleAssignmentEntity, UUID> {
 
     Set<RoleAssignmentEntity> findByActorId(UUID actorId);
 
+    Set<RoleAssignmentEntity> findByActorIdAndRoleType(UUID actorId, String roleType);
+
     void deleteByActorId(UUID actorId);
 
+    @Query(value = "select ra.* from role_assignment ra where ra.actor_id  = :actorId and "
+                   + "role_type = :roleType and ra.attributes ->> 'caseId' = :caseId ;", nativeQuery = true)
+    Set<RoleAssignmentEntity> findByActorIdAndCaseId(String actorId, String caseId, String roleType);
+
+    @Query(value = "select ra.* from role_assignment ra where "
+                   + "role_type = :roleType and ra.attributes ->> 'caseId' = :caseId ;", nativeQuery = true)
```

Comment:
I think, we should place the caseId condition first as it can drastically reduce the result set. maybe roleType is not relevant with caseId but we can keep that as second condition.

Reply:
Updated the condition.